### ISSUE TITLE
Partial chunk fill bug 1.14

### ISSUE THREAD

**Issue by roopekoo** [Brettflan/WorldBorder#145]
I have encountered a bug that when I have a big worldborder, radius over 2k, the filling process leaves unfinished chunks (shaped black squares) that are visible in dynmap. 

I am using:
Paper 1.14.2 build 96
Worldborder 1.9.10 (beta)
Dynmap 3.0 beta-4

How to reproduce:
1. Set the border to 5000x5000 preferably to round. (/wb world set 5000 5000 spawn) (/wb wshape world round)
2. Start the filling process and wait. (/wb world fill)
3. When finished do dynmap fullrender and wait. (/dynmap fullrender world)
4. Profit

Dynmap doesn't render unfinished chunks. And if I go to the black area in dynmap and do updaterender, it doesn't do anything or do radiusrender, it renders only the tiny area to unrendered area where the player currently is. So, the chunks that were unfinished are now rendered and complete when the player visits the chunk.

I currently have around 20k worldborder fully filled and fullrendered. There were no issues in the process of filling and rendering.
Here is an image of the full 20k radius world with those squares from dynmap:
![20k](https://user-images.githubusercontent.com/6078487/59975121-1c707380-95bd-11e9-9493-75f294676cc0.png)
One pixel on the map corresponds to 21m in the game(1:21)
These chunk gaps are about 6 chunks wide.

**Comment by leroy627** [comment]
I had this happen to me as well, a workaround would be to force loading for all chunks by doing `/wb world fill 20 208 true`
Did you ever step into that world before running WB fill by any chance?

**Comment by roopekoo** [comment]
Thank you @leroy627 for suggesting that. With forcing chunks it updates the partial chunks and dynmap renders them.
Still this is an issue for normal usage for huge maps without forcing chunks.

**Comment by Nekuroi** [comment]
Have the same issue, my guess is there is some kind of padding added to already generated chunks because of how wb ignore borders of already explored zones.

https://imgur.com/a/6loIEgq

**Comment by leroy627** [comment]
The 'bug' was introduced as a result of #101 

Looking at the first screenshot it seems as if the black borders appear at a certain interval of chunks loaded(i.e. the area loaded, notice how the 'loaded' areas thin out then grow at the end), which is odd since the code looks like it shouldn't have that happening.

**Comment by Leonardo-Fiori** [comment]
That happens also to me, the dev has stop answering since some time, hope this plugin is not going to be discontinued..

**Comment by mxs42** [comment]
Are you sure that it is in fact is not being generated? 
Because when I went to black areas they were fully generated but Dynmap refused to render them even after update / fullrender.

**Comment by Leonardo-Fiori** [comment]
@mxs42 yeah, they do not generate. That are not simply an empty padding.. if you travel over them they do generate

**Comment by mxs42** [comment]
@Leonardo-Fiori sadly yes, I just tested again and can confirm that they are not being generated :(

**Comment by Tomcatchriss** [comment]
Same here in Paper 1.14.3 and also Spigot 1.14.3

**Comment by Leonardo-Fiori** [comment]
Solution: force the generation using the latest argument.. /wb fill world freq pad true

The last "true" argument forces generation and worked for me

**Comment by JonMagon** [comment]
@Leonardo-Fiori but it will regenerate all chunks? It may take a long time if the world is big.

**Comment by Leonardo-Fiori** [comment]
No, i did it on my world and it is still here, if it regenerated it from scratch I would have killed myself lol

The other main problem now are nether portals.. I had to keep worldborder disabled to make them work. And enable it when expanding map

**Comment by JonMagon** [comment]
Stop, did you find a solution to make nether portals work?

**Comment by Leonardo-Fiori** [comment]
It's not a solution, at least personally I would not call it solution.. disable worldborder plugin and set the vanilla worldborder at the same spot to prevent players from going outside. Then when you need to trim or fill enable it back and do it.. then disable. If you want to keep it enabled find a way to enable to all the players the bypass mode.. wb bypass

**Comment by JonMagon** [comment]
@Leonardo-Fiori I've fixed this for me. Please check if it works.
[WorldBorder-JonMagon_fork0.1.zip](https://github.com/Brettflan/WorldBorder/files/3362812/WorldBorder-JonMagon_fork0.1.zip)

**Comment by Tomcatchriss** [comment]
> Solution: force the generation using the latest argument.. /wb fill world freq pad true
> 
> The last "true" argument forces generation and worked for me

It needs much time to fill and the fullrender takes much more time now. The flat Map looks good. Now, it's rendering the surface. But at the moments it looks good.

**Comment by Gaboros** [comment]
> I had this happen to me as well, a workaround would be to force loading for all chunks by doing `/wb world fill 20 208 true`
> Did you ever step into that world before running WB fill by any chance?

This won't wipe the building we made or redo the modifications we made to the world, does it?

**Comment by cooperisdunn** [comment]
> > I had this happen to me as well, a workaround would be to force loading for all chunks by doing `/wb world fill 20 208 true`
> > Did you ever step into that world before running WB fill by any chance?
> 
> This won't wipe the building we made or redo the modifications we made to the world, does it?

No.

**Comment by Leonardo-Fiori** [comment]
Today we decided to expand the map to 6000 blocks. Starting from the previous 2000 blocks. Its taking ages, i have to keep my server on whitelist for thw whole day.. of course because i had to force the generation using "true". This is barely usable. Please fix :( @Brettflan

**Comment by ishland** [comment]
The same issue in Paper 1.14.4

**Comment by TheoRetisch1** [comment]
I have the same problem under Paper 1.15.2 - 226 with the Worldborder version 1.9.10. The error pattern is not as consistent anymore: https://imgur.com/a/M2mOFGP. But the error description and the solution approach still apply 1:1.
